core/iwasm/aot/aot_loader.c

```diff
@@ -3632,6 +3632,21 @@ load_relocation_section(const uint8 *buf, const uint8 *buf_end,
     return ret;
 }
 
+#if WASM_ENABLE_MEMORY64 != 0
+static bool
+has_module_memory64(AOTModule *module)
+{
+    /* TODO: multi-memories for now assuming the memory idx type is consistent
+     * across multi-memories */
+    if (module->import_memory_count > 0)
+        return !!(module->import_memories[0].memory_flags & MEMORY64_FLAG);
+    else if (module->memory_count > 0)
+        return !!(module->memories[0].memory_flags & MEMORY64_FLAG);
+
+    return false;
+}
+#endif
```

Comment:
In check_feature_flags function of this file, we check whether the related feature is enabled when the feature flag is set in the AOT file. Had better also emit the memory64 feature flag in the AOT file when there is a 64-bit memory in the wasm file, and add the check in check_feature_flags.

Reply:
hmm, I'm quite sure I've got `wasm_memory_check_flags` call in AOT (that's why I refactored it with the `is_aot` flag parameter) but looks like I must have removed it when doing some refactoring. I'll add it back in the `load_memory_info`, I don't think we need to add feature flags if the information about the memory type is already encoded in the aot bundle anyway.